### ISSUE TITLE
add block production

### ISSUE THREAD

**Issue by michele-nuzzi** [HarmonicLabs/pluts-emulator#1]

**Comment by RSoulatIOHK** [comment]
Should this include block sizing and tx validation or would those be in a separate ticket?

**Comment by michele-nuzzi** [comment]
yes there should be at least some tx validation, in particular the checks dependent on the ledger state (the input utxos being available)

the rest of tx validation (balanced tx, plutus, etc...) should be added instead on tx submission, so a separate ticket

**Comment by amnambiar** [comment]
PR is Merged
